runtime/vm/BytecodeInterpreter.hpp

```diff
@@ -5221,12 +5221,17 @@ class INTERPRETER_CLASS
 				UDATA result = preparePinnedVirtualThreadForUnmount(_currentThread, object, true);
 				VMStructHasBeenUpdated(REGISTER_ARGS);
 				if (J9_OBJECT_MONITOR_OOM != result) {
+					PORT_ACCESS_FROM_JAVAVM(_vm);
 					restoreInternalNativeStackFrame(REGISTER_ARGS);
 					/* Handle the virtual thread Object.wait call.
 					 * VirtualThread.timeout is a private field used by both VM and JCL to temporarily hold
 					 * the value of expected wait/park time before a wake up task is scheduled using the value.
 					 */
 					J9VMJAVALANGVIRTUALTHREAD_SET_TIMEOUT(_currentThread, _currentThread->threadObject, millis + (nanos / 1000000));
+					_currentThread->currentContinuation->startTicks = j9time_nano_time();
+					omrthread_monitor_t monitor = _currentThread->currentContinuation->objectWaitMonitor->monitor;
+					/* Trigger MonitorWait hook. */
+					TRIGGER_J9HOOK_VM_MONITOR_WAIT(_vm->hookInterface, _currentThread, monitor, millis + (nanos / 1000000), 0);
```

Comment:
the last parameter is for `nanos`. why are we adding `nanos` to `millis`?
```suggestion
					TRIGGER_J9HOOK_VM_MONITOR_WAIT(_vm->hookInterface, _currentThread, monitor, millis, nanos);
```

Reply:
the JCL code which is responsible for timeout management will always use the rounded millisecond, hence I want to match the actual timer and MONITOR_WAITED event as well.

Reply:
This is not "rounding" it is "truncating" (e.g. 4.9 ms is treated as 4.0 ms).